azurelinuxagent/ga/exthandlers.py

```diff
@@ -969,38 +968,53 @@ def report_ext_handlers_status(self, incarnation_changed=False):
 
         self.write_ext_handlers_status_to_info_file(vm_status)
 
-    @staticmethod
-    def write_ext_handlers_status_to_info_file(vm_status):
-        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE)
+    def write_ext_handlers_status_to_info_file(self, vm_status):
+
+        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE.format(self.last_etag))
 
         agent_details = {
             "agent_name": AGENT_NAME,
-            "current_version": str(CURRENT_VERSION),
             "goal_state_version": str(GOAL_STATE_AGENT_VERSION),
-            "distro_details": "{0}:{1}".format(DISTRO_NAME, DISTRO_VERSION),
-            "last_successful_status_upload_time": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()),
             "python_version": "Python: {0}.{1}.{2}".format(PY_VERSION_MAJOR, PY_VERSION_MINOR, PY_VERSION_MICRO),
             "crp_supported_features": [name for name, _ in get_agent_supported_features_list_for_crp().items()],
-            "extension_supported_features": [name for name, _ in get_agent_supported_features_list_for_extensions().items()]
+            "extension_supported_features": [name for name, _ in
+                                             get_agent_supported_features_list_for_extensions().items()]
         }
 
-        # Convert VMStatus class to Dict.
-        data = get_properties(vm_status)
+        if vm_status is not None:
+            # Convert VMStatus class to Dict.
+            data = get_properties(vm_status)
+            goal_state_status = data['vmAgent']['vm_artifacts_aggregate_status']['goal_state_aggregate_status']
+
+            agent_details["daemon_version"] = data['vmAgent']['version']
+            agent_details["distro_details"] = "{0}:{1}".format(data['vmAgent']['osname'], data['vmAgent']['osversion'])
+            agent_details["agent_status"] = data['vmAgent']['status']
+            agent_details["agent_message"] = data['vmAgent']['message']
+            agent_details["agent_hostname"] = data['vmAgent']['hostname']
+
+            goal_state_aggregate_status = {
+                "processed_time": time.strftime("%Y-%m-%dT%H:%M:%SZ", goal_state_status['_GoalStateAggregateStatus__utc_timestamp']),  # the timestamp is slightly different
```

Comment:
You're accessing a private member of the `GoalStateAggregateStatus` class. Consider using this property instead - 

```python
@property
    def processed_time(self):
        return self.__utc_timestamp
```

Reply:
Good point, Since in the original code 'vm_status' is converted to a dictionary called 'data' I could no longer accesses class properties of GoalStateAggregateStatus. Is there any other good way to do it ?